File: android_common/src/main/java/com/alexstyl/specialdates/events/peopleevents/EventType.java
package com.alexstyl.specialdates.events.peopleevents;

import android.support.annotation.ColorRes;
import android.support.annotation.DrawableRes;

import com.alexstyl.resources.Strings;
import com.alexstyl.specialdates.events.database.EventTypeId;

public interface EventType {

    String getEventName(Strings strings);

    @ColorRes
    int getColorRes();

    @EventTypeId
    int getId();

    @DrawableRes
    int getIconResId();

    int getAndroidType();
}

File: memento/src/main/java/com/alexstyl/specialdates/events/peopleevents/EventType.kt
package com.alexstyl.specialdates.events.peopleevents

import com.alexstyl.specialdates.Strings
import com.alexstyl.specialdates.events.database.EventTypeId

interface EventType {

    fun getEventName(strings: Strings): String

    @get:EventTypeId
    val id: Int
}
